pkg/daemon/update.go

```diff
@@ -1988,6 +1990,77 @@ func (dn *Daemon) cancelSIGTERM() {
 	}
 }
 
+func (dn *Daemon) SetLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
+		},
+	}
+	dn.logSystem("Setting labels %v on node %s", labels, dn.name)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		dn.logSystem("Error in setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	_, err = dn.kubeClient.CoreV1().Nodes().Patch(
+		context.TODO(), dn.name, types.MergePatchType, patchData, metav1.PatchOptions{})
+	if err != nil {
+		dn.logSystem("Error setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	return nil
+}
+
+func (dn *Daemon) setExcludeFromLBLabel(addLabel bool) error {
+	var labelValue interface{}
+	var err error
+	if addLabel {
+		labelValue = ""
```

Comment:
Does the feature allow us to set a value to explain why we have set this label?

Reply:
We could indeed set a value... The cloud service controller and in the az code only check for the presence of this label, without looking at the value.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L673
https://github.com/kubernetes/kubernetes/blob/b68063fce7abf62c69c257610d93ed3eb100956a/staging/src/k8s.io/legacy-cloud-providers/azure/azure.go#L850